backend/matching-service/src/services/rabbitmq.service.ts

```diff
@@ -262,6 +276,42 @@ class RabbitMQConnection {
     createId(userId: string, timestamp: string): string {
         return userId + timestamp
     }
+
+    async listenToDeadLetterQueue() {
+        try {
+            if (!this.channel) {
+                await this.connect()
+            }
+
+            // DeadLetter-Queue setup parameters
+            const DLX_QUEUE = 'deadletter-queue'
+            const DLX_EXCHANGE = 'dlx'
+            const DLX_ROUTING_KEY = 'dlx-key'
+            await this.channel.assertExchange(DLX_EXCHANGE, 'direct', { durable: true })
+            await this.channel.assertQueue(DLX_QUEUE, { durable: true })
+            await this.channel.bindQueue(DLX_QUEUE, DLX_EXCHANGE, DLX_ROUTING_KEY)
+
+            // Consume messages
+            await this.channel.consume(DLX_QUEUE, (msg) => {
+                if (msg) {
+                    const userId = msg.properties.headers.userId
+                    const queue = msg.properties.headers['x-first-death-queue']
+                    logger.info(`[DeadLetter-Queue] Received dead letter message from user ${userId} from ${queue}`)
+                    this.channel.ack(msg)
+                }
+            })
+        } catch (error) {
+            logger.error('[DeadLetter-Queue] Error while consuming from the dead letter queue:', error)
+        }
+    }
+
+    private getDlxArgs() {
```

Comment:
Perhaps can consider directly creating an object that can be used in the argument instead of using a get method to retrieve the same object

Reply:
Works too! But I did this so that the variables will be in a smaller scope (only for the methods that require it) instead of in the entire class 😆